workflow/executor/executor.go

```diff
@@ -543,6 +548,16 @@ func (we *WorkflowExecutor) InitDriver(art wfv1.Artifact) (artifact.ArtifactDriv
 		}
 		return &driver, nil
 	}
+	if art.GCS != nil {
+		credsJSONData, err := we.GetSecrets(we.Namespace, art.GCS.CredentialsSecret.Name, art.GCS.CredentialsSecret.Key)
```

Comment:
I should be getting the secret from a volume mount here in order to bring it in line with the other artifact handlers